cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,502 @@
+        for path in base.rglob(rglob):
+            # Exclude log directory:
+            if path.relative_to(base).parts[0] != 'log':
+                yield path
+
+
+def get_reference_rst(checks):
+    """Print a reference for checks to be carried out.
+
+    Returns:
+        RST compatible text.
+    """
+    output = ''
+    current_checkset = ''
+    for check, meta in checks.items():
+        # Check if the purpose has changed - if so create a new
+        # section title:
+        if meta['purpose'] != current_checkset:
+            current_checkset = meta['purpose']
+            title = CHECKS_DESC[meta["purpose"]]
+            output += f'\n{title}\n{"-" * len(title)}\n\n'
+
+        # Fill a template with info about the issue.
+        template = (
+            '{checkset}{index:003d}\n^^^^\n{summary}\n\n'
+        )
+        if meta['url'].startswith('http'):
+            url = meta['url']
+        else:
+            url = URL_STUB + meta['url']
+        summary = meta.get("rst", meta['short'])
+        msg = template.format(
+            title=check.pattern.replace('\\', ''),
+            checkset=meta['purpose'],
+            summary=summary,
+            url=url,
+            index=meta['index'],
+        )
+        output += msg
+    output += '\n'
+    return output
+
+
+def get_reference_text(checks):
+    """Print a reference for checks to be carried out.
+
+    Returns:
+        RST compatible text.
+    """
+    output = ''
+    current_checkset = ''
+    for check, meta in checks.items():
+        # Check if the purpose has changed - if so create a new
+        # section title:
+        if meta['purpose'] != current_checkset:
+            current_checkset = meta['purpose']
+            title = CHECKS_DESC[meta["purpose"]]
+            output += f'\n{title}\n{"-" * len(title)}\n\n'
+
+        # Fill a template with info about the issue.
+        template = (
+            '{checkset}{index:003d}:\n    {summary}\n\n'
+        )
+        if meta['url'].startswith('http'):
+            url = meta['url']
+        else:
+            url = URL_STUB + meta['url']
+        msg = template.format(
+            title=check.pattern.replace('\\', ''),
+            checkset=meta['purpose'],
+            summary=meta['short'],
+            url=url,
+            index=meta['index'],
+        )
+        output += msg
+    output += '\n'
+    return output
+
+
+def get_option_parser() -> COP:
+    parser = COP(
+        COP_DOC,
+        argdoc=[COP.optional(('DIR ...', 'Directories to lint'))],
+    )
+    parser.add_option(
+        '--inplace', '-i',
+        help=(
+            'Modify files in place, adding comments to files. '
+            'If not set, the script will work as a linter'
+        ),
+        action='store_true',
+        default=False,
+    )
+    parser.add_option(
+        '--ruleset', '-r',
+        help=(
+            'Set of rules to use: '
+            '("728", "style", "all")'
+        ),
+        default='728',
```

Comment:
Response to verbal review from @datamel 

```suggestion
        default='all',
```